### PR TITLE
Hosting: Update Dialog for phpMyAdmin Password Reset

### LINKED ISSUE #95889: phpMyAdmin restore password modal style is off
### Quick summary

When resetting the password for phpMyAdmin, the style of the confirmation modal is off:
* Buttons with wrong border radius
* Modal missing a close button
* Inconsistent padding/margin

Image

### Steps to reproduce

1. Have a site on Business or higher plan.
2. [Activate hosting features](https://wordpress.com/support/activate-your-business-plan/#activate-hosting-access) if they're not yet activated.
3. Go to "Server Settings" tab.
4. Click "resetting the password" link in the "Database access" box.

### What you expected to happen

Modal shown with correct style.

### What actually happened

Modal shown with incorrect style.

### Impact

Some (< 50%)

### Available workarounds?

There is no user impact

### If the above answer is "Yes...", outline the workaround.

_No response_

### Platform (Simple and/or Atomic)

Atomic

### Logs or notes

_No response_

### PR DESCRIPTION
**Pull request by Aurorum**
Fixes #95889 

## Proposed Changes

* Updates the "Reset password" confirmation dialog for the phpMyAdmin card. 

## Why are these changes being made?

The current dialog design is inconsistent with others. Most importantly, the buttons are the wrong way round, which can cause some confusion, but some of the spacing is also off. In addition, we have a `ConfirmDialog` component specifically for this purpose, so it makes sense to use it. 

## Testing Instructions

* Go to `/hosting-config/site.com` on an Atomic site 
* Navigate to Server Settings
* Click "Try resetting the password" under the phpMyAdmin card
* Confirm the new dialog works

| Before | After |
|--------|--------|
| Screenshot 2024-10-31 at 23 06 44 | Screenshot 2024-10-31 at 23 01 43 | 

cc @nightnei, @wongasy, @fushar

### CODE REVIEW

**Comment by wongasy** [comment]
Thanks @Aurorum for putting in a fix! We'll need specs from the design team to ensure the modal component renders with a consistent look in all use cases. I'll get back to you once we get the specs.

**Comment by wongasy** [comment]
My apologies, it looks like I spoke too early! I noticed that you're leverage an existing `<ConfirmDialog>`, which should give us a consistent look. I'll double check with the team.

**Comment by wongasy** [comment]
@Aurorum I checked with the design team. We are actually using the correct [`<Dialog>` component](https://github.com/Automattic/wp-calypso/tree/trunk/packages/components/src/dialog) in production. 

To fix the issue for the phpMyAdmin password reset modal, we need to:
* Update the styles in the `<Dialog>` component
* Switch the order of the buttons so the primary action sits on the right, as you had pointed out
* Add the `showCloseIcon` property to ensure the "x" button shows up

**Comment by Aurorum** [comment]
Ah, thanks for checking! It looks like the issue with spacing was caused by some bleeding styles which I've tidied up here. 

Screenshot 2024-11-02 at 11 14 34

I've also reversed the button order and added a close icon.

File: client/blocks/site-address-changer/style.scss [hunk 1]
```diff
@@ -5,14 +5,13 @@
...
 	max-width: 800px;
 
-}
-
-.dialog__action-buttons {
-	border: 0;
-	padding: 35px 48px 48px;
+	.dialog__action-buttons {
+		border: 0;
+		padding: 35px 48px 48px;
```
Comment (wongasy) [inline client/blocks/site-address-changer/style.scss]: @Aurorum Looks like we'll break the style of `<SiteAddressChanger>` with this change. Here's how you'd test the site address changer: https://wordpress.com/support/changing-site-address/#change-your-word-press-com-site-address

|before|after|
|-------|-----------------|
|Screen Shot 2024-11-04 at 8 51 24 AM|Screen Shot 2024-11-04 at 8 51 19 AM|

To we can customize the button style for this dialog box, we may consider:

* Wrapping the `<Dialog>` component for the `<SiteAddressChanger> [here](https://github.com/Automattic/wp-calypso/blob/d011f3b3dd669d52a300b3923faced62280f0154/client/blocks/site-address-changer/index.jsx#L554) with a div, assign a custom class name to it.

* Updating `<Dialog>` to apply the custom class name to the [<ButtonBar>](https://github.com/Automattic/wp-calypso/blob/d011f3b3dd669d52a300b3923faced62280f0154/packages/components/src/dialog/index.tsx#L100).

**Review by wongasy — CHANGES_REQUESTED** [review]

File: client/blocks/site-address-changer/style.scss (hunk 1 above)
Comment (Aurorum) [reply client/blocks/site-address-changer/style.scss]: Thanks @wongasy - very good catch! I'd appreciate your guidance here on whether we even need these additional styles though. If the aim is consistency of Dialogs across Calypso, I feel like `<SiteAddressChanger/>` might work best without the forced extra padding?

Screenshot 2024-11-05 at 16 58 25

I've committed it for now, but let me know and I can change it as necessary.

Comment (wongasy) [reply client/blocks/site-address-changer/style.scss]: Ah, is the screenshot you shared the outcome of removing `className="site-address-changer"`? If so, I think it's the right call to to stick with the base style from `<Dialog>`!

Comment (Aurorum) [reply client/blocks/site-address-changer/style.scss]: Ah, perfect - yep, the screenshot is with the class name removed and just the default Dialog styles. :)

**Review by wongasy — APPROVED** [review]
@Aurorum The changes work as expected!

|phpMyAdmin Password Reset|Site Address Change|
|-----------------------------|----------------------|
|Screen Shot 2024-11-05 at 3 41 43 PM|Screen Shot 2024-11-05 at 3 39 39 PM|

If you could add the `showCloseIcon` property to the dialog box of `<SiteAddressChanger>`, we're good to go!

**Comment by Aurorum** [comment]
Thanks @wongasy - all done!

**Comment by wongasy** [comment]
@Aurorum I was planning to deploy this today, but it looks like we have a merge conflict. Could you resolve it and I'll deploy the change today!

**Comment by wongasy** [comment]
@Aurorum Hello there! Following up again to see if you could resolve the merge so I can deploy this fix, thanks!!

**Comment by Aurorum** [comment]
oh dear something has gone quite wrong there...let me try this one again! And apologies for the pings to the codeowners...

**Comment by davemart-in** [comment]
No worries! It happens!

**Pull request closed without merging**